### PR TITLE
Products, and Category model creation

### PR DESCRIPTION
**Pull request by redseaworldcoding**

### CODE REVIEW

**Comment by hmtranae** [comment]
category should have @onetomany annotation to users table and
product should have @onetomany annotation to reviews table

File: src/main/java/com/launchacademy/programmingswaglist/models/Category.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    @Column(name = "id", nullable = false, unique = true)
+    private Integer id;
+
+    @NotBlank
+    @Column(name = "name", nullable = false)
+    private String name;
+
+    @ManyToMany(mappedBy = "categories")
+    private Set<Product> product;
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/programmingswaglist/models/Category.java]: please remove extra line here.

File: src/main/resources/db/migration/V3__create_product.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+
+CREATE TABLE products(
+    id SERIAL PRIMARY KEY NOT NULL,
+    name VARCHAR(255) NOT NULL,
+    price NUMERIC(6,2) NOT NULL,
+    description TEXT NOT NULL
+);
+
+
```
Comment (prophetsfall) [inline src/main/resources/db/migration/V3__create_product.sql]: please remove extra line

File: src/main/resources/db/migration/V3__create_product.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
```
Comment (prophetsfall) [inline src/main/resources/db/migration/V3__create_product.sql]: please remove extra lines

File: src/main/java/com/launchacademy/programmingswaglist/models/Product.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    private String description;
+    @ManyToMany
+    @JoinTable(name = "products_categories",
+            joinColumns = @JoinColumn(name = "product_id", referencedColumnName = "id"),
+            inverseJoinColumns = @JoinColumn(name = "category_id", referencedColumnName = "id")
+    )
+    private Set<Category> categories;
+    @OneToMany(mappedBy = "products")
+    private List<Reviews> reviewsList = new ArrayList<>();
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/programmingswaglist/models/Product.java]: please remove extra line.

File: src/main/java/com/launchacademy/programmingswaglist/models/Product.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
+package com.launchacademy.programmingswaglist.models;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/programmingswaglist/models/Product.java]: please remove extra line

File: src/main/resources/db/migration/V3__create_product.sql [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    id SERIAL PRIMARY KEY NOT NULL,
+    name VARCHAR(255) NOT NULL,
+    price NUMERIC(6,2) NOT NULL,
+    description TEXT NOT NULL
+);
+
+CREATE TABLE categories(
+    id SERIAL PRIMARY KEY NOT NULL,
+    name VARCHAR(255) NOT NULL
+
```
Comment (prophetsfall) [inline src/main/resources/db/migration/V3__create_product.sql]: please remove extra line

File: src/main/java/com/launchacademy/programmingswaglist/models/Category.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+package com.launchacademy.programmingswaglist.models;
+
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/programmingswaglist/models/Category.java]: Please remove extra line

File: src/main/java/com/launchacademy/programmingswaglist/models/Category.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/programmingswaglist/models/Category.java]: please remove extra line

**Pull request merged**